src/modules/generation/service.py

```diff
@@ -0,0 +1,174 @@
+import uuid, logging
+from openai import AsyncOpenAI
+from src.settings import settings
+from src.cache import get_cache
+from src.enums import GenerationStatus
+import calendar
+from datetime import datetime, timedelta
+from fastapi import HTTPException, status
+
+
+logger = logging.getLogger(__name__)
+
+
+client = None
+
+
+class GenerationService:
+    def prompt(
+        self,
+        sourceText: str,
+        numberOfQuestions=10,
+    ):
+        return f"""
+    Generate a set of ${numberOfQuestions} difficult educational questions and answers from the following text. I want to use these questions to prepare an examination for my students. Please format the questions and answers as JSON and wrap the JSON response with specific characters for easy parsing in code. Format each question as a question object which comprises of the following five properties;
+
+    1. Question Number. Call this property 'id'. Its value should be the question number
+    2. Question Type. Call this property 'type'. Its value should be the type of question. There are two possible question types, namely 'trueOrFalse' and 'multipleChoice'
+    3. Question. Call this property 'question'. Its value should be the question that the student has to answer based on the text. If the 'type' is 'trueOrFalse', then the value of the 'q' should be a question that students can answer either 'True' or 'False' to. If the 'type' is 'multipleChoice', then the value of the 'q' should be a question that has only one correct answer from a list of options.
+    3. Options. Call this property 'options'. Its value should be a list of string options available for my students to pick from.  If the 'type' is 'trueOrFalse', the options should be 'True' and 'False' because those are the only options available to my students. If the 'type' is 'multipleChoice', the options should be a list of the options my students have to pick from. I want four options only.
+    4. Answer. Call this property 'answers'. Its value should be the correct answer to the question. The value should be the zero-based index of the correct answer in the list of 'options'.
+    5. Context. Call this property 'context'. Its value should be the specific portion of the text that contains the answer to the question. It must read word for word from the text.
+
+    It is important that you add these five properties to each of the question objects. It is equally important that 70% of the questions are of type 'multipleChoice', and '30%' are of type 'trueOrFalse'. Also, ensure that the questions and answers are relevant to the text, and that the answer is correct. Return a JSON object that has one key named questions and the value is a list of question objects with the five previously named properties. Here is the relevant text:
+
+    '''
+    ${sourceText}
+    '''
+
+    """
+
+    def generation_id(self):
```

Comment:
```suggestion
    def get_generation_id(self):
```
I think it should be named like it's an action being done